### PR TITLE
Linux one bm feature

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_bare_metal_server.go [hunk 1]
```diff
@@ -244,9 +244,11 @@ func ResourceIBMIsBareMetalServer() *schema.Resource {
 							Description: "If true, the VPC infrastructure performs any needed NAT operations. If false, the packet is passed unmodified to/from the network interface, allowing the workload to perform any needed NAT operations.",
 						},
 						isBareMetalServerNicInterfaceType: {
-							Type:        schema.TypeString,
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_bare_metal_server.go:247]: Can you please implement the acceptance tests ?
Also can you please attach the test results and the terraform configuration file/code..

Thank you

File: ibm/service/vpc/resource_ibm_is_bare_metal_server.go [hunk 2]
```diff
@@ -905,6 +923,88 @@ func resourceIBMISBareMetalServerCreate(context context.Context, d *schema.Resou
...
+				}
+
+				enableInfraNAT, ok := nic[isBareMetalServerNicEnableInfraNAT]
+				enableInfraNATbool := enableInfraNAT.(bool)
+				if ok {
+					nicobj.EnableInfrastructureNat = &enableInfraNATbool
+				}
+
+				if primaryIpIntf, ok := nic[isBareMetalServerNicPrimaryIP]; ok && len(primaryIpIntf.([]interface{})) > 0 {
+					// primaryIp := primaryIpIntf.([]interface{})[0].(map[string]interface{})
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_bare_metal_server.go:949]: Please remove commented code if they are not relevant and will not be used while releasing..

File: ibm/service/vpc/resource_ibm_is_bare_metal_server_network_interface.go [hunk 3]
```diff
@@ -530,6 +547,109 @@ func createVlanTypeNetworkInterface(context context.Context, d *schema.ResourceD
...
+			reservedIpAutoOk, okAuto := primaryIp[isBareMetalServerNicIpAutoDelete]
+			if okAuto {
+				reservedIpAuto := reservedIpAutoOk.(bool)
+				primaryip.AutoDelete = &reservedIpAuto
+			}
+			if okAdd || okName || okAuto {
+				nicOptions.PrimaryIP = primaryip
+			}
+
+			// reservedIpAddressOk, ok := primaryIp[isBareMetalServerNicIpAddress]
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_bare_metal_server_network_interface.go:611]: same here

File: ibm/service/vpc/resource_ibm_is_bare_metal_server_network_interface.go [hunk 4]
```diff
@@ -715,6 +835,82 @@ func bareMetalServerNICGet(d *schema.ResourceData, meta interface{}, sess *vpcv1
...
+			d.Set(isBareMetalServerNicHref, nic.Href)
+			d.Set(isBareMetalServerNicID, *nic.ID)
+			d.Set(isBareMetalServerNicInterfaceType, *nic.InterfaceType)
+
+			d.Set(isBareMetalServerNicMacAddress, *nic.MacAddress)
+			d.Set(isBareMetalServerNicName, *nic.Name)
+			d.Set(isBareMetalServerNicPortSpeed, nic.PortSpeed)
+
+			primaryIpList := make([]map[string]interface{}, 0)
+			// currentIP := map[string]interface{}{
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_bare_metal_server_network_interface.go:866]: same here

File: ibm/service/vpc/data_source_ibm_is_bare_metal_server.go [hunk 5]
```diff
@@ -558,6 +558,19 @@ func dataSourceIBMISBareMetalServerRead(context context.Context, d *schema.Resou
 				currentPrimNic[isInstanceNicAllowIPSpoofing] = *primNic.AllowIPSpoofing
 				currentPrimNic[isBareMetalServerNicPortSpeed] = *primNic.PortSpeed
 
+				if len(primNic.SecurityGroups) != 0 {
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/data_source_ibm_is_bare_metal_server.go:561]: Please update the documentation and examples for all the datasources and resources accordingly..

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Hi Ujjwal, 
Can you please check on the review comments, Incorporate them if you find them valid and relevant.
Thank you

**Comment by ujjwal-ibm** [comment]
moving to upstream

**Pull request closed without merging**